subsys/mgmt/mcumgr/lib/cmd/os_mgmt/src/os_mgmt.c

```diff
@@ -45,103 +51,167 @@ os_mgmt_echo(struct mgmt_ctxt *ctxt)
 		return MGMT_ERR_EINVAL;
 	}
 
-	err |= cbor_encode_text_stringz(&ctxt->encoder, "r");
-	err |= cbor_encode_text_string(&ctxt->encoder, echo_buf, strlen(echo_buf));
+	err = cbor_encode_text_stringz(&ctxt->encoder, "r")				||
+	      cbor_encode_text_string(&ctxt->encoder, echo_buf, strlen(echo_buf));
 
-	if (err != 0) {
-		return MGMT_ERR_ENOMEM;
-	}
 
-	return 0;
+	return (err == 0) ? 0 : MGMT_ERR_ENOMEM;
 }
 #endif
 
 #if OS_MGMT_TASKSTAT
+static inline int
+os_mgmt_taskstat_encode_thread_name(struct CborEncoder *encoder, int idx,
+				    const struct k_thread *thread)
+{
+	int err = 0;
+
+#ifdef CONFIG_THREAD_NAME
+	ARG_UNUSED(idx);
+	err = cbor_encode_text_string(encoder, thread->name,
+			strnlen(thread->name, CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN - 1));
+#else
+	char thread_name[CONFIG_OS_MGMT_TASKSTAT_THREAD_NAME_LEN];
+
+	thread_name[sizeof(thread_name) - 1] = '0';
```

Comment:
Should this be = 0 literal rather than '0' which is 0x30?

Reply:
Yes, missing slash. Should have just gone with literal 0.

Reply:
Done